lib/mayaUsd/fileio/shading/shadingModeExporterContext.cpp

```diff
@@ -387,13 +394,28 @@ UsdPrim UsdMayaShadingModeExportContext::MakeStandardMaterialPrim(
 
     std::string materialName = name;
     if (materialName.empty()) {
-        MStatus           status;
-        MFnDependencyNode seDepNode(_shadingEngine, &status);
-        if (!status) {
+        std::string sgName;
+
+        MFnDependencyNode fnDepNode;
+        if (fnDepNode.setObject(_shadingEngine) == MS::kSuccess) {
+            sgName = fnDepNode.name().asChar();
+        } else {
             return ret;
         }
-        MString seName = seDepNode.name();
-        materialName = seName.asChar();
+
+        std::smatch match;
+        if (std::regex_match(sgName, match, _templatedRegex)) {
```

Comment:
Would it work if we compare the SG name with the surface shader name? If they only differ by `SG` suffix, we have an original name.

Reply:
That is an optimization that could be done: If both names differ only by the presence of a terminal "SG" the SG name should be taken as is since it is already in sync with the surface. If the DFA generated by std::regex is slow that could save some time.

Reply:
It's not really optimization why I mentioned it. Currently, you hardcoded the node types in template regex. For any shader type that isn't captured in there, we will fail to use the surface shader name. So it's more robustness than optimization that I'm talking about.